rkt/metadata_service.go

```diff
@@ -541,12 +539,19 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
```

Comment:
I'd always imagined there would be some way for subsystems to register cleanup handlers for `rkt gc` to execute as part of `deleting`, something as simple as a directory full of executable entries executed in lexical order might suffice.